mirgecom/boundary.py

```diff
@@ -1141,27 +1158,25 @@ class FarfieldBoundary(PrescribedFluidBoundary):
         \nabla q_{bc} = \nabla q^{-}
 
     .. automethod:: __init__
-    .. automethod:: farfield_state
+    .. automethod:: state_plus
+    .. automethod:: state_bc
     .. automethod:: temperature_bc
+    .. automethod:: grad_cv_bc
+    .. automethod:: grad_temperature_bc
     """
 
-    def __init__(self, numdim, free_stream_pressure,
-                 free_stream_velocity, free_stream_temperature,
-                 free_stream_mass_fractions=None):
+    def __init__(self, dim, free_stream_pressure, free_stream_velocity,
+                 free_stream_temperature, free_stream_mass_fractions=None):
         """Initialize the boundary condition object."""
-        if len(free_stream_velocity) != numdim:
+        if len(free_stream_velocity) != dim:
             raise ValueError("Free-stream velocity must be of ambient dimension.")
```

Comment:
```suggestion
        dim = len(free_stream_velocity)
```

If seems a little odd to force the user to specify `dim` for a check that happens inside the BC object.

Reply:
https://github.com/illinois-ceesd/mirgecom/pull/970/commits/7cc91555139d7b27aaba6d7fee2d14dce230a3af